qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -164,34 +192,61 @@ def load_account(self) -> Optional[AccountProvider]:
         # Prevent edge case where no hubs are available.
         providers = self.providers()
         if not providers:
-            logger.warning('No Hub/Group/Projects could be found for this '
-                           'account.')
+            logger.warning('No Hub/Group/Projects could be found for this account.')
             return None
 
-        return providers[0]
+        # The provider for the default open access project.
+        default_provider = providers[0]
+
+        # If specified, attempt to get the provider stored for the account.
+        if stored_provider_hgp:
+            hub, group, project = stored_provider_hgp.to_tuple()
+            try:
+                default_provider = self.get_provider(hub=hub, group=group, project=project)
+            except IBMQProviderError as ex:
+                raise IBMQAccountError('The default provider (hub/group/project) stored on '
```

Comment:
If you're going to re-raise to add more context, can you keep it as the more specific `IBMQProviderError` (and maybe ditch the `from` as it doesn't add any new information).